### PR TITLE
Ch setup redux 160577562

### PR DESCRIPTION
**Pull request by bevkololi**
**What does this PR do?**

Sets up redux and creates the appropriate folder/file structure for React/Redux workflow

**Description of Task to be completed?**

-Install react-redux and redux-thunk
-Create redux store
-Create root reducer
-Provide redux store to components

**How should this be manually tested?**

`Clone Repo` --> *git clone* 
`Run server` --> *npm start* 

**Any background context you want to provide?**

For redux the actions and reducers folders are required. These will however be created by individuals working on the specific features.

**What are the relevant pivotal tracker stories?**

Create the appropriate folder/file structure for React/Redux workflow
The story an be accessed [here](https://www.pivotaltracker.com/story/show/160577562)

**Screenshots (if appropriate)**

N/A

**Questions:**

N/A

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 import './assets/index.css';
 import App from './containers/App';
 import * as serviceWorker from './serviceWorker';
+import configureStore from './store/index';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const store = configureStore();
+
+ReactDOM.render(
+  <Provider store={store}>
```
[the remark this answers is not in the document]
Comment (paulrimiru) [reply src/index.js:12]: @bevkololi Look into this

File: src/components/Button/index.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import React, { Component } from 'react';
 class Button extends Component {
   render() {
     return (
-      <button>
+      <button type="button">
```
[the remark this answers is not in the document]
Comment (paulrimiru) [reply src/components/Button/index.js:6]: +1

File: src/containers/App/reducer.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const initialState = {
+
```
Comment (paulrimiru) [inline src/containers/App/reducer.js]: Remove empty line here

File: src/store/rootReducer.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { combineReducers } from 'redux';
+import app from '../containers/App/reducer';
+
+
```
Comment (paulrimiru) [inline src/store/rootReducer.js:3]: Remove extra line here

**Review by paulrimiru — CHANGES_REQUESTED** [review]

**Review by mutaimwiti — COMMENTED** [review]
@bevkololi The PR would follow our convections if it was in this format:

#[STORY_ID] [Story description].

Refer to the [engineering-playbook](https://github.com/andela/engineering-playbook/blob/master/5.%20Developing/Conventions/readme.md#pr-title)

File: src/store/rootReducer.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from 'redux';
+import app from '../containers/App/reducer';
```
[the remark this answers is not in the document]
Comment (paulrimiru) [reply src/store/rootReducer.js:2]: @bevkololi please have @charisschomba add this for you

[the remark this answers is not in the document]
Comment (bevkololi) [reply src/store/rootReducer.js:2]: We're working on it @PaulKariukiRimiru

**Pull request closed without merging**